public/models/GameMap.js

```diff
@@ -30,105 +30,117 @@ class GameMap {
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            // take one coin to draw as they are the same
+            coin.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          case 1:
+            wall.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          case 2:
+            goal.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
+              ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
+              this.tileWidth,
+              this.tileHeight,
+            );
+            break;
+          default:
+        }
+      }
     }
+  }
 
-    //Marianna
-    //load map from the file
-    //set goal tile and time limit
-    loadMap (data) {
-        this.tiles = data.tiles;
-        //get random position for goal
-        let [row, column] = [-1, -1];
-        do {
-            row = Utils.getRandomNumber(0, this.tiles.length);
-            column = Utils.getRandomNumber(0, this.tiles[row].length);
-        } while (this.tiles[row][column] !== 0);
-        this.tiles[row][column] = 2;
-        this.goalRow = row;
-        this.goalColumn = column;
-        //set time limit based on size - 1 minute for 10x10
-        this.timeLimit = 60 * 1000 * (this.tiles.length * this.tiles[0].length / 100);
-        //generate coins
-        //5 coins per 10x10 
-        this.generateCoins(this.tiles.length * this.tiles[0].length / 20)
-    }
+  // Marianna
+  // load map from the file
+  // set goal tile and time limit
+  loadMap(mapFile) {
+    const fs = require('fs');
+    const data = fs.readFileSync(mapFile);
+    this.tiles = JSON.parse(data).tiles;
+    // get random position for goal
+    let [row, column] = [-1, -1];
+    do {
+      row = Utils.getRandomNumber(0, this.tiles.length);
+      column = Utils.getRandomNumber(0, this.tiles[row].length);
+    } while (this.tiles[row][column] !== 0);
+    this.tiles[row][column] = 2;
+    this.goalRow = row;
+    this.goalColumn = column;
+    // set time limit based on size - 1 minute for 10x10
+    this.timeLimit = 60 * 1000 * (this.tiles.length * this.tiles[0].length / 100);
+    // generate coins
+    // 5 coins per 10x10
+    this.generateCoins(this.tiles.length * this.tiles[0].length / 20);
```

Comment:
also please explain this calc

Reply:
the upper one just counts the length * width of the map which if we have 50x50 will be 2500 which would result in 2500 minutes for this map, but that is a bit too much. So I divide it by 100 so it will be 25 minutes. If we take it with 10x10 it will be 100 minutes per this size so after division it's 1 minute.
For the lower one, it again takes length*width which for this example 10x10 would be again 100. So per 10x10 map we would have 100 coins which is a bit too much. So I divide it by 20 which will get the result of 5 aka 5 coins for 10x10 map piece. Originally i had there /10 so it would be 10 coins but I lowered it because we will be adding gems and traps so it doesn't get filled up just with coins. We can adjust this number during the development to see what works the best.